### ISSUE TITLE
Not all concurrency annotations added to Objective-C are visible in the Swift translation

### ISSUE THREAD

**Issue by mattmassicotte** [swiftlang/swift#74629]
### Motivation

I've discovered that if I add concurrency-related annotations to Objective-C signatures, the compiler understands them, but Xcode does not make them visible. Here's an example:

```objc
+ (void)jetPackConfiguration:(void (NS_SWIFT_SENDABLE ^)(void))completionHandler NS_SWIFT_DISABLE_ASYNC;
```

Note the lack of `@Sendable`
```swift
class func jetPackConfiguration(_ completionHandler: @escaping () -> Void)
```

Now it is entirely possible this is not a compiler issue. I'm just guessing because of this:

https://github.com/swiftlang/swift/pull/72037

### Proposed solution

I'd like to see the fully annotated function siguature!

**Comment by MaxDesiatov** [comment]
Is this reproducible for you with VS Code?

**Comment by mattmassicotte** [comment]
Yeah, it could be this is an Xcode issue. But, reason I assumed that may not be the case is because it took a compiler change to expose global actors in this same place.

**Comment by hamishknight** [comment]
Just to clarify, where are you viewing this information? Looks like `@Sendable` is printed in the generated interfaces, but it is not printed in quick help. In any case, this information should be coming from SourceKit, so it's fine for the bug to be here.

**Comment by mattmassicotte** [comment]
Sorry, I missed this question! Yes, I'm not seeing these details in Xcode's Quick Help popover window.

I've been experimenting with a bunch of other annotations as well. It seems like `NS_SWIFT_UI_ACTOR` definitely does end up working in this case, but many others do not.
